fix h3 failing queries having const columns

bharatnc opened the pull request:
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fix bug for h3 funcs containing const columns which cause queries to fail. 

Currently queries like the following containing const columns fail:
```
select h3ToParent(641573946153969375, arrayJoin([1,2]));
Code: 44. DB::Exception: Received from localhost:9000. DB::Exception: Illegal type UInt64 of argument 1 of function h3ToParent. Must be UInt64.. (ILLEGAL_COLUMN)
```

@kitaisreal actually fixed it for `geoToH3` in #34731. Fixing this in all other places by converting const columns to full columns and tried adding tests.

Related: #34708

bharatnc committed:
convert const cols to full cols

bharatnc committed:
add tests

bharatnc committed:
fix - skip fast tests

kitaisreal commented:
@bharatnc there is segmentation fault https://s3.amazonaws.com/clickhouse-test-reports/34743/d9acbf480ca5f24af17a57b97e0efbc696075731/fuzzer_astfuzzerubsan,actions//report.html.
```
SELECT NULL, toFloat64('-1'), -2147483648, h3CellAreaRads2(arrayJoin([9223372036854775807, 65535, NULL]))
```

bharatnc committed:
fix fuzzer segfault

bharatnc commented:
> @bharatnc there is segmentation fault https://s3.amazonaws.com/clickhouse-test-reports/34743/d9acbf480ca5f24af17a57b97e0efbc696075731/fuzzer_astfuzzerubsan,actions//report.html.
> 
> ```
> SELECT NULL, toFloat64('-1'), -2147483648, h3CellAreaRads2(arrayJoin([9223372036854775807, 65535, NULL]))
> ```

It happens only in `h3CellAreaRads2`  due to invalid H3 index. Added the check to prevent such segfaults and also added a test. Doesn't happen with `h3CellAreaM2`, but still added the same check just in case.

bharatnc commented:
@kitaisreal this fix might need backporting ?
